### ISSUE TITLE
Channel.get_messages() crashes half the time

### ISSUE THREAD

**Issue by iirelu** [zeyla/serenity#237]
```
thread 'serenity client' panicked at 'called `Result::unwrap()` on an `Err` value: Json(ErrorImpl { code: Message("unknown field `animated`, expected `id` or `name`"), line: 1, column: 4439 })', /checkout/src/libcore/result.rs:916:4
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at /checkout/src/libstd/sys_common/backtrace.rs:68
             at /checkout/src/libstd/sys_common/backtrace.rs:57
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/panicking.rs:381
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:397
   4: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:577
   5: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:538
   6: std::panicking::begin_panic_fmt
             at /checkout/src/libstd/panicking.rs:522
   7: rust_begin_unwind
             at /checkout/src/libstd/panicking.rs:498
   8: core::panicking::panic_fmt
             at /checkout/src/libcore/panicking.rs:71
   9: core::result::unwrap_failed
             at /checkout/src/libcore/macros.rs:23
  10: <core::result::Result<T, E>>::unwrap
             at /checkout/src/libcore/result.rs:782
  11: <stestcase::Handler as serenity::client::event_handler::EventHandler>::on_ready
             at src/main.rs:21
  12: serenity::client::dispatch::handle_event::{{closure}}
             at /home/anna/.cargo/registry/src/github.com-1ecc6299db9ec823/serenity-0.4.5/src/client/dispatch.rs:402
  13: <core::result::Result<T, E>>::map
             at /checkout/src/libcore/result.rs:468
  14: serenity::client::dispatch::handle_event
             at /home/anna/.cargo/registry/src/github.com-1ecc6299db9ec823/serenity-0.4.5/src/client/dispatch.rs:398
  15: serenity::client::dispatch::dispatch
             at /home/anna/.cargo/registry/src/github.com-1ecc6299db9ec823/serenity-0.4.5/src/client/dispatch.rs:63
  16: <serenity::client::bridge::gateway::shard_runner::ShardRunner<H>>::run::{{closure}}
             at /home/anna/.cargo/registry/src/github.com-1ecc6299db9ec823/serenity-0.4.5/src/client/bridge/gateway/shard_runner.rs:123
  17: <F as threadpool::FnBox>::call_box
             at /home/anna/.cargo/registry/src/github.com-1ecc6299db9ec823/threadpool-1.7.1/src/lib.rs:95
  18: threadpool::spawn_in_pool::{{closure}}
             at /home/anna/.cargo/registry/src/github.com-1ecc6299db9ec823/threadpool-1.7.1/src/lib.rs:767
```

Half the time I run get_messages(), it crashes. The other half, it works fine. Repeatedly running it gets a mix of crashes and successes. There's no pattern as far as I can figure out. Here's an adequate test case for experiencing this behaviour:

```rust
extern crate serenity;

use serenity::model::{Ready, Message};
use serenity::prelude::*;
use serenity::client::Context;


fn main() {
    let token = std::env::var("DISCORD_TOKEN").unwrap();
    let mut client = Client::new(&token, Handler);
    client.start();
}

struct Handler;

impl EventHandler for Handler {
    fn on_ready(&self, ctx: Context, r: Ready) {
        let mut server = r.guilds[0].id();
        let channel = server.channels().unwrap().values().next().unwrap().clone();

        let recent_messages = channel.messages(|m| m).unwrap(); // crashes here
    }
}
```

**Comment by zeyla** [comment]
Thanks, I'll look into this.

**Comment by zeyla** [comment]
Could you try out branch `branch-v0.4.6` and see if 5286949f424e824784344ebb7b7af4e52fb819c3 fixes it?

Add the following to your `Cargo.toml` to use it:
```toml
[dependencies.serenity]
branch = "branch-v0.4.6"
git = "https://github.com/zeyla/serenity"
```

**Comment by zeyla** [comment]
Published a fix as [v0.4.6 on crates.io](https://crates.io/crates/serenity), thanks for the report.
